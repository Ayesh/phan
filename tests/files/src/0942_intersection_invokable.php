<?php

namespace NS942;

use Countable;
use Traversable;

interface IntInvokable {
    public function __invoke(int $arg): void {
        var_dump($arg);
    }
}

function test1(Countable $value) {
    if ($value instanceof IntInvokable) {
        var_dump($value());  // missing arg
    }
}

function test2(Countable $value) {
    if ($value instanceof Traversable) {
        $value();  // missing arg
    }
}
