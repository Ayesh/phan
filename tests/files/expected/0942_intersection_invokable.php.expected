%s:16 PhanParamTooFew Call with 0 arg(s) to \NS942\IntInvokable::__invoke(int $arg) which requires 1 arg(s) defined at %s:9
%s:16 PhanTypeVoidArgument Cannot use void return value $value() as a function argument
%s:22 PhanUndeclaredInvokeInCallable Possible attempt to access missing magic method __invoke of '\Countable&\Traversable'
